sonar-scanner-engine/src/main/java/org/sonar/batch/report/MeasuresPublisher.java

```diff
@@ -41,16 +41,16 @@
 
 public class MeasuresPublisher implements ReportPublisherStep {
 
-  private static final class MeasureToReportMeasure implements Function<Measure, BatchReport.Measure> {
+  private static final class MeasureToReportMeasure implements Function<Measure, ScannerReport.Measure> {
     private final BatchComponent resource;
-    private final BatchReport.Measure.Builder builder = BatchReport.Measure.newBuilder();
+    private final ScannerReport.Measure.Builder builder = ScannerReport.Measure.newBuilder();
 
     private MeasureToReportMeasure(BatchComponent resource) {
       this.resource = resource;
     }
 
     @Override
-    public BatchReport.Measure apply(@Nonnull Measure input) {
+    public ScannerReport.Measure apply(@Nonnull Measure input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this "Nonnull" annotation to honor the overridden method's contract. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)